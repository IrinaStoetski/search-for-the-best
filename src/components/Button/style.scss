@import '../../styles/shared/style.scss';

.btn {
  display: flex;
  justify-content: center;
  padding: 8px;
  font-size: 16px;
  border: none;
  border-radius: 16px;
  cursor: pointer;

  &:disabled {
    opacity: 0.6;
  }

  &.primary {
    min-width: 80px;
    min-height: 36px;
    color: #fff;
    background-color: $primary-color;
  }
}
