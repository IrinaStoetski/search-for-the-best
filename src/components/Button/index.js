// @flow
import * as React from 'react';
// Components
import Spinner from '../Spinner';
// Styles
import './style.scss';

type Props = {
    type?: 'primary',
    customClass?: string,
    buttonType?: 'reset' | 'button' | 'submit',
    disabled?: boolean,
    loading?: boolean,
    children?: React.Node,
    onClick?: Function,
  };

const Button = ({
  type = 'primary',
  customClass = '',
  buttonType = 'submit',
  disabled = false,
  onClick,
  loading = false,
  children,
}: Props) => {
  const componentClassName = `btn ${type} ${customClass}`;
  return (
    // eslint-disable-next-line react/button-has-type
    <button
      className={componentClassName}
      type={buttonType}
      disabled={disabled || loading}
      onClick={onClick}
    >
      {loading ? <Spinner /> : children}
    </button>
  );
};


export default Button;
