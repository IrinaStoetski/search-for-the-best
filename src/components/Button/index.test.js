/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';

test('renders correctly simple primary button without props', () => {
  const tree = renderer
    .create(<Button />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
