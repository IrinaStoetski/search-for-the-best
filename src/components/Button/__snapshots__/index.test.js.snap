// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly simple primary button with button type and onclick handler 1`] = `
<button
  className="btn primary "
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`Button component renders correctly simple primary button without props 1`] = `
<button
  className="btn primary "
  disabled={false}
  type="submit"
/>
`;
